workspace.bzl

```diff
@@ -0,0 +1,42 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# instructions to build nlohmann library after downloading it.
+build_file_content_txt = """
+cc_library(
+    name = "nlohmann_json_lib",
+    hdrs = glob([
+        "include/nlohmann/*.hpp",
+        "include/nlohmann/**/*.hpp",
+        "include/nlohmann/*/*/*.hpp",
+    ]),
+    includes = ["nlohmann_json_lib"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "json",
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+    deps = [":nlohmann_json_lib"],
+)"""
+
+# supported versions repositories
+def load_nlohmann_3_11_3():
+    http_archive(
+        name = "nlohmann_ext_3.11.3",
+        urls = ["https://github.com/nlohmann/json/archive/v3.11.3.tar.gz"],
+        sha256 = "0d8ef5af7f9794e3263480193c491549b2ba6cc74bb018906202ada498a79406",
+        build_file_content = build_file_content_txt,
+
+        strip_prefix = "json-3.11.3",
+    )
```

Comment:
Can this be changed if `nlohmann/json` is also built with `bazel`?

Reply:
Not sure what you mean. Bazel downloads all dependencies (in this case nlohmann/json) and starts build process, so downloaded nlohmann/json will be build with bazel.

Reply:
Let's take `vcpkg` design as an example since both projects are defined there. There you have 2 projects:
- [nlohmann-json](https://github.com/microsoft/vcpkg/tree/master/ports/nlohmann-json)
- [json-schema-validator](https://github.com/microsoft/vcpkg/tree/master/ports/json-schema-validator)

The design is that `json-schema-validator` has a dependency on `nlohmann-json` [^1]. That way `nlohmann-json` is not vendored inside `json-schema-validator` and the end-user decides which version of each package they want, defaulting to the latest version if none is defined. The supported version is usually controlled by the metadata of `vcpkg.json` or by build failure if we specify it in the CMake `find_package` (I need to double check what version compatibility `nlohmann/json` advertises).

It would be preferred if `bazel` could have a similar design, but this also depends on how do `bazel` projects work. E.g. how does another project that uses `json-schema-validator` work with this project once the bazel files are included? Is there an equivalent model to `vcpkg` where the build instructions are handled in a separate repo by the packagers?

[^1]: https://github.com/microsoft/vcpkg/blob/2ddc10c8afa00a762a69f21f6427849fe906ede0/ports/json-schema-validator/vcpkg.json#L9-L12